distributed/executor.py

```diff
@@ -193,6 +193,22 @@ def release(self):
             self._cleared = True
             self.executor._dec_ref(tokey(self.key))
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state.update({'event': self.event.is_set(), 'status': self.status})
+        del state['executor']
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        e = default_executor()
+        event = Event()
+        if state.pop('event'):
+            event.set()
+        e.futures[tokey(state['key'])] = {'event': event,
+                                          'status': state.pop('status')}
+        state['executor'] = e
```

Comment:
This could also send the message up to the scheduler of "Hey this executor cares about this key now!"

     e._send_to_scheduler({'op': 'update-graph': 'tasks': {}, 'keys': [tokey(key)], 'client': e.id})